File: src/main/java/org/team1/nbe1_3_team01/domain/attendance/service/AttendanceDeleter.java
package org.team1.nbe1_3_team01.domain.attendance.service;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Transactional;
import org.team1.nbe1_3_team01.domain.attendance.entity.Attendance;
import org.team1.nbe1_3_team01.domain.attendance.service.port.AttendanceRepository;

@Component
@RequiredArgsConstructor
public class AttendanceDeleter {

    private final AttendanceRepository attendanceRepository;

    @Transactional
    public void delete(Attendance attendance) {
        attendanceRepository.deleteById(attendance.getId());
    }
}

File: src/main/java/org/team1/nbe1_3_team01/domain/attendance/service/AttendanceDeleter.kt
package org.team1.nbe1_3_team01.domain.attendance.service

import org.springframework.stereotype.Component
import org.team1.nbe1_3_team01.domain.attendance.entity.Attendance
import org.team1.nbe1_3_team01.domain.attendance.service.port.AttendanceRepository
import org.team1.nbe1_3_team01.global.exception.AppException
import org.team1.nbe1_3_team01.global.util.ErrorCode

@Component
class AttendanceDeleter(
    private val attendanceRepository: AttendanceRepository
) {

    fun delete(attendance: Attendance) {
        if (attendance.id == 0L) {
            throw AppException(ErrorCode.ATTENDANCE_NOT_FOUND)
        }

        attendanceRepository.deleteById(attendance.id)
    }
}
